modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -741,15 +741,18 @@ private function payerData() {
 	 * @throws \WooCommerce\PayPalCommerce\WcGateway\Exception\NotFoundException If a setting was not found.
 	 */
 	private function url(): string {
+		$intent = ( $this->settings->has( 'intent' ) ) ? $this->settings->get( 'intent' ) : 'capture';
```

Comment:
Good catch! done :)

Reply:
@AlexP11223 I reverted `capture` in lowecase on `SmartButton.php` as it is used in the JS script.

I think the reason of letter case inconsistency is because lowecase is used in JS while uppercase is used for API calls:
https://developer.paypal.com/docs/platforms/checkout/add-capabilities/auth-capture/#1-change-approval-intent
https://developer.paypal.com/docs/platforms/checkout/add-capabilities/auth-capture/#2-create-order

So the rule to follow seems to be `intent` setting stored in lowercase and then use `strtoupper` when needed, for example here: https://github.com/woocommerce/woocommerce-paypal-payments/blob/79e2dd2ef5f9b46ce6f6b8a74c28be1bc38f36e4/modules/ppcp-wc-gateway/extensions.php#L73

Reply:
hm, in this example it's unclear why they suddenly decided to convert to upper if non-API values like our settings are supposed to be lower.

But yeah we probably should always do case-insensitive comparison of intents unless we are really sure about case.

Or treat them as completely separate things (because e.g. we can add some other options to our intent settings) but it is difficult to enforce because they are just strings.